libvast/include/vast/plugin.hpp

```diff
@@ -402,6 +403,47 @@ class rest_endpoint_plugin : public virtual plugin {
     = 0;
 };
 
+// -- loader plugin -----------------------------------------------------
+
+using input_loader = std::function<auto()->generator<chunk_ptr>>;
+using input_parser = std::function<auto(chunk_ptr)->generator<table_slice>>;
+using options = record;
+
+/// A loader plugin transfers input data into a stream of chunks.
+/// @relates plugin
+class loader_plugin : public virtual plugin {
+public:
+  /// Returns the loader.
+  [[nodiscard]] virtual auto make_loader(options, operator_control_plane&) const
+    -> caf::expected<input_loader>
+    = 0;
+
+  /// Returns the default parser for this loader.
+  [[nodiscard]] virtual auto
+  make_default_parser(options, operator_control_plane&) const
+    -> caf::expected<input_parser>
+    = 0;
+};
+
+class stdin_loader_plugin : public virtual loader_plugin {
+public:
+  static constexpr inline auto max_chunk_size = size_t{16384};
+
+  [[nodiscard]] auto make_loader(options, operator_control_plane&) const
+    -> caf::expected<input_loader> override;
+
+  [[nodiscard]] auto make_default_parser(options, operator_control_plane&) const
+    -> caf::expected<input_parser> override;
+
+  [[nodiscard]] caf::error
+  initialize(const record& plugin_config, const record& global_config) override;
+
+  [[nodiscard]] std::string name() const override;
+
+private:
+  std::chrono::milliseconds read_timeout{vast::defaults::import::read_timeout};
+};
```

Comment:
This should be a builtin rather than directly in the libvast, otherwise this defeats the purpose of the plugin API.